### ISSUE TITLE
Want support for certain modifications instead of having to `terraform destroy` everything

### ISSUE THREAD

**Issue by askfongjojo** [oxidecomputer/terraform-provider-oxide#82]
Currently the provider doesn't support resource updates because most resources do not support PUTs. However, a number of modifications already have API support:

higher priority:
- [x] instance: attach/detach disks
- [x] instance: add/delete NIC, change primary NIC

lower priority:
- [x] vpc: modify description, dns_name
- [x] subnet: modify description (after #79)
